detectors/aws/eks/eks_resource_detector.go

```diff
@@ -0,0 +1,243 @@
+		return nil, err
+	}
+
+	// Return empty resource object if not running in EKS
+	if !isEks {
+		return resource.Empty(), nil
+	}
+
+	// Create variable to hold resource attributes
+	labels := []label.KeyValue{}
+
+	// Get clusterName and append to labels
+	clusterName, err := getClusterName(detector.utils)
+	if err != nil {
+		return nil, err
+	}
+	if clusterName != "" {
+		labels = append(labels, semconv.K8SClusterNameKey.String(clusterName))
+	}
+
+	// Get containerID and append to labels
+	containerID, err := detector.utils.getContainerID()
+	if err != nil {
+		return nil, err
+	}
+	if containerID != "" {
+		labels = append(labels, semconv.ContainerIDKey.String(containerID))
+	}
+
+	// Return new resource object with clusterName and containerID as attributes
+	return resource.New(labels...), nil
+
+}
+
+// isEks checks if the current environment is running in EKS
+func isEks(utils detectorUtils) (bool, error) {
+	if !isK8s(utils) {
+		return false, nil
+	}
+
+	// Make HTTP GET request
+	awsAuth, err := utils.fetchString("GET", k8sSvcURL+authConfigmapPath)
+	if err != nil {
+		return false, fmt.Errorf("isEks() error: %s", err.Error())
+	}
+
+	return awsAuth != "", nil
+}
+
+// isK8s checks if the current environment is running in a Kubernetes environment
+func isK8s(utils detectorUtils) bool {
+	return utils.fileExists(k8sTokenPath) && utils.fileExists(k8sCertPath)
+}
+
+// fileExists checks if a file with a given filename exists
+// this function implements the detectorUtils interface
+func (eksUtils eksDetectorUtils) fileExists(filename string) bool {
+	info, err := os.Stat(filename)
+	if os.IsNotExist(err) {
+		return false
+	}
+	return !info.IsDir()
+}
+
+// fetchString executes an HTTP request with a given HTTP Method and URL string
+// this function implements the detectorUtils interface
+func (eksUtils eksDetectorUtils) fetchString(httpMethod string, URL string) (string, error) {
+
+	// Create new HTTP request object
+	request, err := http.NewRequest(httpMethod, URL, nil)
+	if err != nil {
+		return "", fmt.Errorf("failed to create new HTTP request with method=%s, URL=%s", httpMethod, URL)
+	}
+
+	// Set HTTP request header with authentication credentials
+	authHeader, err := getK8sCredHeader()
+	if err != nil {
+		return "", err
+	}
+	request.Header.Set("Authorization", authHeader)
+
+	// Get certificate
+	caCert, err := ioutil.ReadFile(k8sCertPath)
+	if err != nil {
+		return "", fmt.Errorf("failed to read file with path %s", k8sCertPath)
+	}
+	caCertPool := x509.NewCertPool()
+	caCertPool.AppendCertsFromPEM(caCert)
+
+	// Set HTTP request timeout and add certificate
+	client := &http.Client{
+		Timeout: millisecondTimeOut * time.Millisecond,
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				RootCAs: caCertPool,
+			},
+		},
+	}
+
+	// Execute HTTP request
+	response, err := client.Do(request)
+	if err != nil {
+		return "", fmt.Errorf("failed to execute HTTP request with method=%s, URL=%s", httpMethod, URL)
+	}
+
```

Comment:
Should there be a response status check here?  Is a `200` response expected, or would anything `2xx` be ok?